draft-hendrickson-privacypass-public-metadata.md

```diff
@@ -29,58 +29,309 @@ author:
+Given this configuration and these inputs, the two messages exchanged in
+this protocol are described below. This section uses notation described in
+{{POPRF, Section 4}}, including SerializeElement and DeserializeElement,
+SerializeScalar and DeserializeScalar, and DeriveKeyPair.
 
+The constants `Ne` and `Ns` are as defined in {{POPRF, Section 4}} for
+OPRF(P-384, SHA-384). The constant `Nk`, which is also equal to `Nh` as defined
+in {{POPRF, Section 4}}, is defined in {{iana}}.
 
-# Terminology
+## Client-to-Issuer Request {#private-request}
 
-{::boilerplate bcp14}
+The Client first creates a context as follows:
 
-The following terms are used throughout this document.
+~~~
+client_context = SetupPOPRFClient("P384-SHA384", pkI)
+~~~
 
-- Public Metadata: Information that can be viewed by the Client, Attester, Issuer, and Origin, and cryptographically bound to a token.
+Here, "P384-SHA384" is the identifier corresponding to the
+OPRF(P-384, SHA-384) ciphersuite in {{POPRF}}. SetupPOPRFClient
+is defined in {{POPRF, Section 3.2}}.
 
-# Notation
+The Client then creates an issuance request message for a random value `nonce`
+with the input challenge and Issuer key identifier as described below:
 
-The following terms are used throughout this document to describe the protocol operations in this document:
+~~~
+nonce = random(32)
+challenge_digest = SHA256(challenge)
+token_input = concat(0xDA7B, // Token type field is 2 bytes long
+                     nonce,
+                     challenge_digest,
+                     token_key_id)
+blind, blinded_element, tweaked_key = client_context.Blind(token_input, extensions, pkI)
+~~~
 
- - len(s): the length of a byte string, in bytes.
- - concat(x0, ..., xN): Concatenation of byte strings. For example, concat(0x01, 0x0203, 0x040506) = 0x010203040506
- - int_to_bytes: Convert a non-negative integer to a byte string. int_to_bytes is implemented as I2OSP as described in {{Section 4.1 of RFC8017}}. Note that these functions operate on byte strings in big-endian byte order.
+The Blind function is defined in {{POPRF, Section 3.3.3}}.
+If the Blind function fails, the Client aborts the protocol.
+The Client stores the `nonce`, `challenge_digest`, and `tweaked_key` values locally
+for use when finalizing the issuance protocol to produce a token
+(as described in {{private-finalize}}).
+
+The Client then creates an ExtendedTokenRequest structured as follows:
+
+~~~
+struct {
```

Comment:
Is it assumed that an issuer endpoint will only support extended or non-extended? Otherwise, the issuer may have trouble parsing this differently from a TokenRequest - the first bytes will still be the TokenRequest

Reply:
Oh, good point! This needs a new media type since it's a new resource.